src/main/java/org/openlmis/integration/dhis2/domain/element/DataElement.java

```diff
@@ -15,62 +15,67 @@
 
 package org.openlmis.integration.dhis2.domain.element;
 
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.CascadeType;
 import javax.persistence.Column;
 import javax.persistence.Entity;
 import javax.persistence.FetchType;
 import javax.persistence.JoinColumn;
 import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
 import javax.persistence.Table;
-import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
 import lombok.Setter;
 import lombok.ToString;
 import org.javers.core.metamodel.annotation.TypeName;
 import org.openlmis.integration.dhis2.domain.BaseEntity;
 import org.openlmis.integration.dhis2.domain.dataset.Dataset;
+import org.openlmis.integration.dhis2.domain.schedule.Schedule;
 
 @Entity
 @TypeName("DataElement")
 @Table(name = "data_element", schema = "dhis2")
+@Getter
+@Setter
 @NoArgsConstructor
-@AllArgsConstructor
+@RequiredArgsConstructor
```

Comment:
Isn't it inconsistent with @NoArgsConstructor annotation?

Reply:
Annotation @NoArgsConstructor is being used there to create new Entity from DTO object.
Annotation @RequiredArgsConstructor is being used in normal aplication flow and for testing purposes.
I think both are required here.

Reply:
If you need an argument constructor in your entity, maybe consider using the **AllArgsConstructor** annotation, like in [FacilityType](https://github.com/OpenLMIS/openlmis-referencedata/blob/master/src/main/java/org/openlmis/referencedata/domain/FacilityType.java) or [User](https://github.com/OpenLMIS/openlmis-referencedata/blob/master/src/main/java/org/openlmis/referencedata/domain/User.java)

Reply:
DataElement has bidirectional relationship with Schedule. In it's class there is a reference to list of all schedules connected with this element. We don't want this list to be set from the constructor, so AllArgsConstructor is not applicable here.